spectractor/extractor/dispersers.py

```diff
@@ -699,57 +725,57 @@ def load_specs(self, verbose=True):
 
         The files do not exist:
 
-        >>> h = Hologram(label='XXX')
-        >>> h.N((500,500))
-        350
-        >>> h.theta((500,500))
-        0
-        >>> h.holo_center
-        [1024.0, 1024.0]
+        >>> h = Hologram(label='XXX')  # doctest: +ELLIPSIS
+        Traceback (most recent call last):
+        ...
+        FileNotFoundError:...
 
         """
         if verbose:
             self.my_logger.info(f'\n\tLoad disperser {self.label}:\n\tfrom {os.path.join(self.data_dir, self.label)}')
-        filename = self.data_dir + self.label + "/hologram_grooves_per_mm.txt"
+        filename = os.path.join(self.data_dir, self.label, "hologram_grooves_per_mm.txt")
         if os.path.isfile(filename):
             a = np.loadtxt(filename)
             self.N_x, self.N_y, self.N_data = a.T
             if parameters.CCD_REBIN > 1:
                 self.N_x /= parameters.CCD_REBIN
                 self.N_y /= parameters.CCD_REBIN
-            N_interp = interpolate.interp2d(self.N_x, self.N_y, self.N_data, kind='cubic')
+            self.N_interp = interpolate.interp2d(self.N_x, self.N_y, self.N_data, kind='cubic')
             self.N_fit = fit_poly2d(self.N_x, self.N_y, self.N_data, order=2)
-            self.N_interp = lambda x: float(N_interp(x[0], x[1]))
         else:
             self.is_hologram = False
             self.N_x = np.arange(0, parameters.CCD_IMSIZE)
             self.N_y = np.arange(0, parameters.CCD_IMSIZE)
-            filename = self.data_dir + self.label + "/N.txt"
+            filename = os.path.join(self.data_dir, self.label, "N.txt")
             if os.path.isfile(filename):
                 a = np.loadtxt(filename)
-                self.N_interp = lambda x: a[0]
-                self.N_fit = lambda x, y: a[0]
+
+                def N_func(x, y):
+                    return a[0]
+                self.N_interp = N_func
+                self.N_fit = N_func
```

Comment:
A few things here:
1) I don't know if this will pickle, but we'll find out. It's probably fine, I'm just not sure how different this actually will be to using a `lambda`.
2) The function takes two parameters, ignores both of them, and just returns the first line of the file regardless? That seems odd.

Reply:
It is odd but the default is that it depends on two parameters. So to remain generic I define this dummy function with two parameters that always return the same value for N or theta